src/wallet/db.cpp

```diff
@@ -755,9 +740,11 @@ std::string BerkeleyDatabaseVersion()
 
 void BerkeleyDatabase::Release()
 {
-    {
-        LOCK(cs_db);
-        --env->mapFileUseCount[strFile];
-    }
+    m_refcount--;
```

Comment:
In commit "walletdb: track database file use as m_refcount within BerkeleyDatabase" (9603e233c9c4049a267282615b5e4f31600656e6)

Even though `m_refcount` is atomic you still have to lock cs_db while updating it, otherwise the `m_db_in_use.notify_all()` call may result in a lost wakeup and infinite hang if the `m_db_in_use.wait()` lambda happens to run at the same time `m_refcount` is decremented and return `false` (see https://stackoverflow.com/a/41875321 or https://github.com/isocpp/CppCoreGuidelines/issues/554).

Reply:
Done